### ISSUE TITLE
New style pipelined streaming responses break

### ISSUE THREAD

**Issue by ahopkins** [sanic-org/sanic#2198]
**Describe the bug**
The second request of pipelined HTTP requests will fail if it is a new-style streaming response.

**Code snippet**
```python
@app.route("/simple")
def simple_handler(request: Request):
    return text("simple")


@app.route("/new")
async def new_handler(request: Request):
    resp = await request.respond()
    await resp.send("new")
    await resp.eof()


@app.route("/old")
async def old_handler(request: Request):
    async def do_stream(resp):
        await resp.write("old")

    return stream(do_stream)
```

Executes fine with:

```bash
$ curl localhost:9999/simple localhost:9999/old
simple
old
```

The chunks sent are:

```
b'HTTP/1.1 200 OK\r\ncontent-length: 6\r\nconnection: keep-alive\r\ncontent-type: text/plain; charset=utf-8\r\n\r\nsimple'
b'HTTP/1.1 200 OK\r\ntransfer-encoding: chunked\r\nconnection: keep-alive\r\ncontent-type: text/plain; charset=utf-8\r\n\r\n3\r\nold\r\n'
b'0\r\n\r\n'
```

It breaks down with new style streaming responses:

```bash
$ curl localhost:9999/simple localhost:9999/new
simple
curl: (18) transfer closed with 3 bytes remaining to read
new
```

We never get to those last three bytes because there is an exception:

```
b'HTTP/1.1 200 OK\r\ncontent-length: 6\r\nconnection: keep-alive\r\ncontent-type: text/plain; charset=utf-8\r\n\r\nsimple'
b'HTTP/1.1 200 OK\r\ncontent-length: 6\r\nconnection: keep-alive\r\ncontent-type: text/plain; charset=utf-8\r\n\r\nnew'
[2021-07-14 23:48:57 +0300] [1906838] [ERROR] Exception occurred while handling uri: 'http://localhost:9999/new'
Traceback (most recent call last):
  File "/home/adam/Projects/Sanic/sanic/sanic/app.py", line 777, in handle_request
    response = await response
  File "/home/adam/Documents/Writing/Sanic/Chapters/05/src/sse/server.py", line 26, in new_handler
    await resp.eof()
  File "/home/adam/Projects/Sanic/sanic/sanic/response.py", line 246, in eof
    await self.send("", True)
  File "/home/adam/Projects/Sanic/sanic/sanic/response.py", line 122, in send
    await self.stream.send(data, end_stream=end_stream)
  File "/home/adam/Projects/Sanic/sanic/sanic/http.py", line 381, in http1_response_normal
    raise ServerError("Response was smaller than content-length")
sanic.exceptions.ServerError: Response was smaller than content-length
```

**Expected behavior**
Both `/new` and `/old` should behave the same

**Environment (please complete the following information):**
 - Version `main`

**Additional context**

Without digging too deeply, I suspect the reason is that the subsequent request is using the same self.response_func as the first instead of redefining it.

**Comment by ahopkins** [comment]
Without digging too deeply, I suspect the reason is that the subsequent request is using the same `self.response_func` as the first instead of redefining it.

**Comment by ahopkins** [comment]
@Tronic, thoughts?

**Comment by Tronic** [comment]
The new response incorrectly says `content-length: 6` where it should be using chunked encoding instead, and the true length is 3 anyway. Can you investigate from where it obtains this header?

Note: AFAIK ALL of those test cases should be using chunked encoding because the sizes are not known when the first chunk of the body is written. Manually specifying the content-length header when size is known and/or specifying `res.send(data, end_stream=True)` is preferred for maximal performance, and is supported for this reason. However, I cannot see from where it could obtain the value 6, other than the word "simple" perhaps by accident being this length.

**Comment by ahopkins** [comment]
@Tronic Pinging you on Discord re: this

**Comment by ahopkins** [comment]
Fixed in #2201
